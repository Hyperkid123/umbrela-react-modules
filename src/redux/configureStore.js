import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
