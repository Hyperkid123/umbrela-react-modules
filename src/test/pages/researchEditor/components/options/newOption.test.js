import React from 'react'
import NewOption, {NewOption as Snapshot} from '../../../../../pages/researchEditorModule/components/options/newOption';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('New option component', () => {
  const initialState  = {
    questions: {
      activeQuestion: {
        questionId: 1
      }
    },
    options: {
      options: []
    },
    ui: {
      draggingElement: false
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<NewOption store={store}/>);
    expect(wrapper).toBeDefined();
  })
  it('should render correclty', () => {
      const tree = shallow(
        <Snapshot
          activeQuestion={{questionId: 1}}
          options={[
            {optionOrder: 1, optionId: 1, optionType: 'ColumnOption'},
            {optionOrder: 10, optionId: 10, optionType: 'RowOption'}
          ]}
          translate={jest.fn()}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  it('should synchronizeOption correctly', () => {
    const synchronizeOption = jest.fn();
    const wrapper = shallow(
      <Snapshot
        activeQuestion={{questionId: 1}}
        options={[
          {optionOrder: 1, optionId: 1, optionType: 'ColumnOption'},
          {optionOrder: 10, optionId: 10, optionType: 'RowOption'}
        ]}
        synchronizeOption={synchronizeOption}
        translate={jest.fn()}
      />
    );
    wrapper.setState({optionText: 'foo'});
    wrapper.instance().createOption();
    expect(synchronizeOption.mock.calls.length).toEqual(1);
  })

  it('should set option text corrcetly', () => {
    const wrapper = shallow(
      <Snapshot
        activeQuestion={{questionId: 1}}
        options={[
          {optionOrder: 1, optionId: 1, optionType: 'ColumnOption'},
          {optionOrder: 10, optionId: 10, optionType: 'RowOption'}
        ]}
        translate={jest.fn()}
      />
    );
    wrapper.instance().setOptionText('foo');
    expect(wrapper.state()).toEqual({optionText: 'foo'});

  })
});
