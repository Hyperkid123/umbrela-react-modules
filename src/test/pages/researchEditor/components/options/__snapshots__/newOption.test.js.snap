// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New option component should render correclty 1`] = `
<div
  className="sc-bdVaJa fQfuvz"
>
  <styled.div
    column={true}
  >
    <TextField
      fullWidth={true}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Nová možnost"
      required={false}
      select={false}
      value=""
    />
    <TextFieldComent
      alignRight={true}
      error={false}
      label="0 z 255 znaků"
    />
  </styled.div>
  <WithStyles(IconButton)
    onClick={[Function]}
    tooltip="Přidat možnost"
  >
    <pure(Add) />
  </WithStyles(IconButton)>
</div>
`;
